{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8001    54\n",
       "8002    54\n",
       "8003    72\n",
       "8004    69\n",
       "8005    60\n",
       "8006    39\n",
       "8007    51\n",
       "8008    66\n",
       "8009    72\n",
       "8010    54\n",
       "8011    54\n",
       "8012    72\n",
       "8013    63\n",
       "8014    69\n",
       "8015    48\n",
       "8016    54\n",
       "8017    72\n",
       "8018    54\n",
       "8019    75\n",
       "8020    57\n",
       "8021    60\n",
       "8022    54\n",
       "8023    72\n",
       "8024    69\n",
       "8025    42\n",
       "8026    48\n",
       "8027    66\n",
       "8028    60\n",
       "8029    66\n",
       "8030    48\n",
       "        ..\n",
       "8472    54\n",
       "8473    63\n",
       "8474    75\n",
       "8475    66\n",
       "8476    54\n",
       "8477    60\n",
       "8478    63\n",
       "8479    57\n",
       "8480    63\n",
       "8481    42\n",
       "8482    45\n",
       "8483    54\n",
       "8484    51\n",
       "8485    60\n",
       "8486    66\n",
       "8487    57\n",
       "8488    45\n",
       "8489    57\n",
       "8490    30\n",
       "8491    36\n",
       "8492    66\n",
       "8493    33\n",
       "8494    54\n",
       "8495    48\n",
       "8496    57\n",
       "8497    72\n",
       "8498    57\n",
       "8499    48\n",
       "8500    57\n",
       "8501    63\n",
       "Name: JUNCTION_LENGTH, Length: 500, dtype: int64"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "# READ MAKEDB TAB FILE FOR EACH SAMPLE / PATIENT\n",
    "\n",
    "df=pd.read_table('resultsHC_38_BB_genotyped_db-pass-props.tab')\n",
    "\n",
    "#GET V-D-J FAMILY,GENE,ALLELE CALLS (SINGLE ASSINGMENT OR FIRST MENTIONED)\n",
    "\n",
    "df['V_GENE'] = df['V_CALL'].str.split('*',1).str[0]\n",
    "df['V_ALLELE']=df['V_CALL'].str.split(',',1).str[0]\n",
    "df['V_FAMILY'] = df['V_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['D_GENE'] = df['D_CALL'].str.split('*',1).str[0]\n",
    "df['D_ALLELE']=df['D_CALL'].str.split(',',1).str[0]\n",
    "df['D_FAMILY'] = df['D_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['J_GENE'] = df['J_CALL'].str.split('-',1).str[0]\n",
    "df['J_FAMILY'] = df['J_CALL'].str.split('*',1).str[0]\n",
    "\n",
    "df\n",
    "\n",
    "#REMOVE NONFUNCTIONAL GENES (\"OR\")\n",
    "\n",
    "df2=df[~df.V_CALL.str.contains('OR')]\n",
    "\n",
    "#REMOVE NONFUNCETIONAL SEQUENCES \n",
    "# STOP CODONS \n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "# READ MAKEDB TAB FILE FOR EACH SAMPLE / PATIENT\n",
    "\n",
    "df=pd.read_table('resultsHC_38_BB_genotyped_db-pass-props.tab')\n",
    "\n",
    "#GET V-D-J FAMILY,GENE,ALLELE CALLS (SINGLE ASSINGMENT OR FIRST MENTIONED)\n",
    "\n",
    "df['V_GENE'] = df['V_CALL'].str.split('*',1).str[0]\n",
    "df['V_ALLELE']=df['V_CALL'].str.split(',',1).str[0]\n",
    "df['V_FAMILY'] = df['V_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['D_GENE'] = df['D_CALL'].str.split('*',1).str[0]\n",
    "df['D_ALLELE']=df['D_CALL'].str.split(',',1).str[0]\n",
    "df['D_FAMILY'] = df['D_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['J_GENE'] = df['J_CALL'].str.split('-',1).str[0]\n",
    "df['J_FAMILY'] = df['J_CALL'].str.split('*',1).str[0]\n",
    "\n",
    "df\n",
    "\n",
    "#REMOVE NONFUNCTIONAL GENES (\"OR\")\n",
    "\n",
    "df2=df[~df.V_CALL.str.contains('OR')]\n",
    "\n",
    "#REMOVE NONFUNCETIONAL SEQUENCES \n",
    "# STOP CODONS \n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "# READ MAKEDB TAB FILE FOR EACH SAMPLE / PATIENT\n",
    "\n",
    "df=pd.read_table('resultsHC_38_BB_genotyped_db-pass-props.tab')\n",
    "\n",
    "#GET V-D-J FAMILY,GENE,ALLELE CALLS (SINGLE ASSINGMENT OR FIRST MENTIONED)\n",
    "\n",
    "df['V_GENE'] = df['V_CALL'].str.split('*',1).str[0]\n",
    "df['V_ALLELE']=df['V_CALL'].str.split(',',1).str[0]\n",
    "df['V_FAMILY'] = df['V_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['D_GENE'] = df['D_CALL'].str.split('*',1).str[0]\n",
    "df['D_ALLELE']=df['D_CALL'].str.split(',',1).str[0]\n",
    "df['D_FAMILY'] = df['D_CALL'].str.split('-',1).str[0]\n",
    "\n",
    "df['J_GENE'] = df['J_CALL'].str.split('-',1).str[0]\n",
    "df['J_FAMILY'] = df['J_CALL'].str.split('*',1).str[0]\n",
    "\n",
    "df\n",
    "\n",
    "#REMOVE NONFUNCTIONAL GENES (\"OR\")\n",
    "\n",
    "df2=df[~df.V_CALL.str.contains('OR')]\n",
    "\n",
    "#REMOVE NONFUNCETIONAL SEQUENCES \n",
    "\n",
    "# STOP CODONS \n",
    "#df2=df2[~df2.STOP.str.contains('TRUE') == True]\n",
    "df2\n",
    "# INDELS \n",
    "#df2=df2[~df2.INDELS.str.contains('TRUE')]\n",
    "\n",
    "# IN_FRAME\n",
    "#df2=df2[~df2.IN_FRAME.str.contains('TRUE')]\n",
    "\n",
    "#FUNCTIONAL \n",
    "#df2=df2[~df2.FUNCTIONAL.str.contains('TRUE')]\n",
    "\n",
    "#JUNCTION \n",
    "#AA TRANSLATION IN FRAME  OF THE SEQUENCE DEMENDS MODULO 0 WHEN DIVIDING BY 3\n",
    "\n",
    "df2=df2[df2.JUNCTION_LENGTH%3==0]\n",
    "\n",
    "#check \n",
    "np.unique(df2['JUNCTION_LENGTH']).shape\n",
    "np.unique(df['JUNCTION_LENGTH']).shape # there are only 33 unique values which means we can use it or try to - as a feature@\n",
    "\n",
    "#AA TRANSLATION OF THE SEQUENCE DEMENDS MORE THEN 3 AA \n",
    "df2=df2[df2.JUNCTION_LENGTH>=12]\n",
    "#df2.info()\n",
    "\n",
    "#REMOVE NA's in JUNCTION \n",
    "#df2=df2[~df2.JUNCTION_LENGTH=='NA']\n",
    "df2['JUNCTION_LENGTH'][8000:8500]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/reut/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Alphabet import generic_dna\n",
    "\n",
    "\n",
    "df2['JUNCTION_AA']=df2['JUNCTION'].apply(lambda x: Seq(x, generic_dna))\n",
    "\n",
    "df2['JUNCTION_AA'][1:8000]= df2['JUNCTION_AA'][1:8000].apply(lambda x: x.translate())\n",
    "\n",
    "\n",
    "\n",
    "#df2['JUNCTION_AA'] = df2['JUNCTION'].translate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
